src/training/flava_eval/finetune_glue.py

```diff
@@ -211,6 +264,47 @@ def compute_metrics(model, dataloader, device, args):
     metrics["loss"] = val_loss / samples_seen
     return metrics
 
+def train_n_steps(model, data, optimizer, scheduler, early_stop, device, args):
+    model.train()
+    progress_bar = tqdm(total=args.num_steps)
+    data_train = iter(data['train'])
+    for i in range(args.num_steps):
+        scheduler(i)
+
+        try:
+            batch = next(data_train)
+        except StopIteration:
+            data = get_task_dataloaders(args)
+            data_train = iter(data['train'])
+            batch = next(data_train)
+        text = batch["text"].to(device)
+        label = batch["label"].to(device)
+
+        logits = model(text)
+        if args.task_name == "stsb":
+            logits = logits.view(-1)
+            label = label.view(-1).float()
+        loss = get_loss_fn(args.task_name)(logits, label)
```

Comment:
you don't need to `get_loss_fn` every training step. outside of the loop you can just do `loss_fn = get_loss_fn(args.task_name)` and then here you can do `loss = lss_fn(logits, labels)`